full-stack-practice/client/src/lib/read.ts

```diff
@@ -0,0 +1,9 @@
+import { products, type Product } from './data';
+
+export function readCatalog(): Promise<Product[]> {
+  return Promise.resolve(products);
```

Comment:
I was hoping you'd reimplement these to make fetch calls. You have your fetch calls in your components, but it's a cleaner design to put them all in a file like this.